sbt

```diff
@@ -117,6 +117,16 @@ download_url () {
   } && [[ -f "$jar" ]]
 }
 
+user_home () {
+  local user_home
+  user_home=$("$java_cmd" -XshowSettings:properties -version 2>&1 | sed -n 's/\s*user.home = //p' 2>/dev/null)
```

Comment:
will this be called on every startup, or only when sbt is not installed yet?

Reply:
It's executed from within a shell function, so: only when there's a need for resolving a path under user's home directory.

There are currently 2 cases: either to form a path where to download the launcher jar (if needed), or to sync "preloaded" jars from sbt's installation directory to user home directory (if needed).

Reply:
Each call to `time java -XshowSettings:properties -version` adds around 100ms.

For `--client` mode that could potentially double our triple the startup - https://github.com/sbt/sbt-launcher-package/pull/334

Reply:
Do we measure this anywhere?  It would have been nice if https://github.com/sbt/sbt-launcher-package/pull/334 also added tests for this

Reply:
No currently we don't have any benchmark or measurements on `sbt` script startup time.

Reply:
OK, following above concern, I've amended the commit with a second approach that factors out property search.